core/src/components/button/button.tsx

```diff
@@ -153,19 +153,27 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
    */
   @Event() ionBlur!: EventEmitter<void>;
 
-  connectedCallback(): void {
-    // Allow form to be submitted through `ion-button`
-    if (this.type !== 'button' && hasShadowDom(this.el)) {
```

Comment:
I removed the `hasShadowDom` check. Since we are running this in the `render` function. The Shadow DOM is already created by the time this code runs.